File: tests/fixtures/java_gradle_sample/src/main/java/com/example/myapp/infrastructure/UserRepo.java
package com.example.myapp.infrastructure;

import com.example.myapp.domain.User;
import java.util.List;

/**
 * UserRepo is the infrastructure implementation for user storage.
 */
public class UserRepo {
    public User findUser(int id) {
        return new User(id, "Bob");
    }

    public List<User> findAll() {
        return List.of(new User(1, "Alice"), new User(2, "Bob"));
    }
}

File: tests/fixtures/kotlin_sample/src/infrastructure/UserRepo.kt
package com.example.kotlinsample.infrastructure

import com.example.kotlinsample.domain.User

/**
 * UserRepo is the infrastructure implementation for user storage.
 */
class UserRepo {
    fun findUser(id: Int): User = User(id, "Bob")
    fun findAll(): List<User> = listOf(User(1, "Alice"), User(2, "Bob"))
}
